### DISCUSSION TITLE
Remote key attestation and revoking trust

### DISCUSSION THREAD

**Discussion by freedge** [pkic/community#32] [Crypto]
Sometimes backdoors are found in HSM products such as CVE-2015-5464 where privileged admins are able to extract keys. Those keys could have been previously "confirmed" with the "cmu getpkc" command.
It seems the keys originating from these HSM could be cloned into newer, not vulnerable HSM, but if at any point the key had been on a vulnerable HSM then it cannot be trusted.
Do you know how revocation is done on remote key attestations in such case? (also asking on https://github.com/ThalesGroup/luna-pkc-validator/issues/1)

**Comment by vanbroup** [comment]
Key attestation is essentially a snapshot of specific attributes **at a given point in time**. These attributes can include firmware versions, timestamps, freshness nonces, and other relevant details. The exact attributes included in an attestation can vary, making standardization efforts crucial for consistency and reliability.

One such effort is the [IETF draft on X.509-based attestation evidence](https://datatracker.ietf.org/doc/draft-ounsworth-rats-x509-evidence/), which aims to standardize how these attestations are structured and verified. This draft helps ensure that key attestations are trustworthy and interoperable across different systems and platforms.
